### PR TITLE
DC-3310 - fix acknowledgement reference length with EIS submission

### PR DESCRIPTION
**Pull request by elabeca**

### CODE REVIEW

**Review by alexflav23 — APPROVED** [review]

File: app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala [hunk 1]
```diff
@@ -124,8 +124,13 @@ class SecureMessageService @Inject()(
   private def forwardMessage(conversationId: String, messagesRequest: CustomerMessage)(
     implicit ec: ExecutionContext,
     request: Request[_]): EitherT[Future, SecureMessageError, Unit] = {
+    val ACKNOWLEDGEMENT_REFERENCE_MAX_LENGTH = 32
```
Comment (OlegEfrem) [inline app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala:127]: why you need max 32 where on line 134 govuk-tax is added ?

Comment (elabeca) [reply app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala:127]: I'm getting a validation error when testing in FIT1 with that reference being more than 32 characters.

Comment (OlegEfrem) [reply app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala:127]: does X-Correlation-ID have the same prefix you're adding to the requestId: govuk-tax- ?

Comment (elabeca) [reply app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala:127]: no, this is about `acknowledgementReference`, which needs to be 32 char max, I didn't change the code for requestId, that's existing code for another story. I think maybe the auditing story Rob worked on.

**Pull request merged**